opm/output/eclipse/Summary.cpp

```diff
@@ -102,18 +103,51 @@ measure rate_unit< rt::gas >() { return measure::gas_surface_rate; }
 template<> constexpr
 measure rate_unit< Phase::GAS >() { return measure::gas_surface_rate; }
 
-template< rt phase >
+template< rt phase, bool injection = true >
 inline quantity rate( const fn_args& args ) {
     double sum = 0.0;
 
     for( const auto* ecl_well : args.ecl_wells ) {
-        if( args.wells.wells.count( ecl_well->name() ) == 0 ) continue;
-        sum += args.wells.at( ecl_well->name() ).rates.get( phase, 0.0 );
+        const auto& name = ecl_well->name();
+        if( args.wells.wells.count( name ) == 0 ) continue;
+        const auto v = args.wells.at( name ).rates.get( phase, 0.0 );
+        if( ( v > 0 ) == injection )
```

Comment:
There are four states here `{ ( v >  0) , ( v < 0 )} x { true , false }` - but I assume only `{(v > 0) , true }` and `{ (v < 0) , false }` are valid - otherwise you have a massive internal error? So I guess you just do `v += sum`?